src/craftdef.cpp

```diff
@@ -229,6 +229,7 @@ static void craftDecrementOrReplaceInput(CraftInput &input,
 				rep.deSerialize(j->second, gamedef->idef());
 				item.remove(1);
 				found_replacement = true;
+				pairs.erase(j);
```

Comment:
the way it is right now is not safe while iterating the list
bad:
```cpp
for (auto j = pairs.begin(); j != pairs.end(); j++) {
  if (any condition)
    pairs.erase(j);
}
```
good:
```cpp
for (auto j = pairs.begin(); j != pairs.end(); ) {
  if (any condition)
    j = pairs.erase(j);
  else
    j++;
}
```

Reply:
I think it's fine, since the loop breaks right after erasing.

Reply:
@sfan5 your good example is also incorrect. `std::vector` invalidates any iterator after `erase` call, because `vector::iterator` points to the `vector::data`, and vector resizing can result in reallocation.

This `erase` is fine, because of `break` later.